### DISCUSSION TITLE
Partially-handled escape sequences displaying random characters

### DISCUSSION THREAD

**Discussion by ahwatts** [wezterm/wezterm#4351] [Q&A]
### What Operating System(s) are you running on?

Linux Wayland

### Which Wayland compositor or X11 Window manager(s) are you using?

I'm using whatever the default is on Fedora 38 (and thus probably GNOME 44), but I think it's Mutter? There's a process under gnome-shell (a sibling of XWayland) called `mutter-x11-frames`, so I assume that's what it is.

```
awatts@ironic-mullet ~ $ Xwayland -version
The X.Org Foundation Xwayland Version 22.1.9 (12201009)
X Protocol Version 11, Revision 0
Build ID: xorg-x11-server-Xwayland 22.1.9-2.fc38
awatts@ironic-mullet ~ $ mutter --version
mutter 44.5
```

### WezTerm version

wezterm 20230712-072601-f4abf8fd

### Ask your question!

I've noticed that sometimes when I type things, some random gibberish characters show up after what I type. They look like characters from escape sequences that I guess weren't escaped or were ignored. So far I've noticed two cases where they seem to be happening, and both are pretty obscure, so I decided to post this as a question rather than a bug report, because it's possible that this is just weird behavior associated with a whole bunch of different terminal emulators trying to fool one another into behaving properly. FWIW, this does not happen when I use GNOME Terminal, though.

In both cases, I have to be SSHed into a remote system.

 I'm not sure if this is a property of the specific remote system or not, or of remote systems in general, or of any terminfo I'm setting when SSHing.

The first is if I then run `emacs` in that SSH session, open a Ruby file, and type a comment. an additional "2~" (two tilde) will appear after the hash sign starting the comment. Here's a screencast:

[Screencast from 2023-09-27 16-30-07.webm](https://github.com/wez/wezterm/assets/308758/2c90594b-e473-44ff-8883-31edfdf6e8cb)

I assume this is left over from some font locking sequence that was run in the terminal but not fully interpreted.

The second is that if I start up `tmux` and then, within the `tmux` session, run the `mycli` MySQL client (a fancier Python alternative to the standard MySQL CLI client). If I then type "select count(*)" after I type the ")", a bunch of junk shows up:

[Screencast from 2023-09-27 16-51-47.webm](https://github.com/wez/wezterm/assets/308758/724676b1-6a24-4166-b5c2-af230ad9b54c)

This doesn't happen if I run `mycli` outside of `tmux`, or if I run `mycli` locally, either inside or outside of `tmux`. Only ssh -> tmux -> mycli seems to make that happen. Again, this looks like a bunch of syntax highlighting gone mad.

I'm happy to try help figure out what's ultimately being sent to wezterm that's causing the issue. I'm not sure what would help in figuring out what's going on, though.

**Comment by vollink** [comment]
Just another user here.  FWIW, I've seen this behavior in iTerm2 when I turned on `CSI u mode` (which surrounds modified keys with escape sequences), meaning even a capitol letter (shifted) would end up with errant garbage on the command line.  CSI u mode is great for the FEW applications that support it, but it wreaks havoc on everything else.  It's just my guess.

Check here: https://wezfurlong.org/wezterm/config/key-encoding.html#xterm-modifyotherkeys
